src/test/java/com/example/music_player/service/SourceServiceTest.java

```diff
@@ -51,105 +57,82 @@ public void setup() {
         testArray = "testByteArray".getBytes();
         testBoolean = false;
         sourceList = List.of(source1, source2);
+        inputStream = new ByteArrayInputStream(StandardCharsets.UTF_16.encode("testString").array());
     }
 
     @Test
-    void save() {
-
+    void saveWhenSourceDoNotExist() throws IOException {
+        when(sourceRepository.isExistByNameAndFileType(song.getName()
+                , mockMultipartFile.getContentType()))
+                .thenReturn(testBoolean);
+        when(storageSourceService.save(
+                mockMultipartFile.getInputStream()
+                , mockMultipartFile.getName()
+                , mockMultipartFile.getContentType()))
+                .thenReturn(sourceList);
+        doNothing().when(sourceRepository).save(source1);
+
+        sourceService.save(mockMultipartFile, song, song.getId());
+        assertEquals(testBoolean, false);
+        verify(storageSourceService, times(1))
+                .save(mockMultipartFile.getInputStream()
+                        , mockMultipartFile.getName()
+                        , mockMultipartFile.getContentType());
+        verify(sourceRepository, times(1)).save(source1);
     }
 
     @Test
-    void findByName() {
-        when(sourceRepository.findByNameAndStorageType(anyString(), anyString(), anyString()))
-                .thenReturn(source1);
-        assertEquals(sourceRepository.findByNameAndStorageType(source1.getName(), source1.getStorage_types(), source1.getFileType()), source1);
-        verify(sourceRepository, times(1)).findByNameAndStorageType(anyString(), anyString(), anyString());
-    }
-
-    @Test
-    void setStorageType() {
-        source1.setStorage_types(testStorage_type);
-        assertEquals(source1.getStorage_types(), testStorage_type);
-    }
+    void saveWhenSourceExist() throws IOException {//second branch of save() method
+        lenient().when(sourceRepository.isExistByNameAndFileType(song.getName()
+                , mockMultipartFile.getContentType()))
+                .thenReturn(true);
+
+        IllegalStateException exception = assertThrows(IllegalStateException.class, () ->
+                        sourceService.save(mockMultipartFile, song, song.getId()),
+                "source do not fined");
+        String expectedMessage = "do not fined";
+        Assertions.assertTrue(exception.getMessage().contains(expectedMessage));
 
-    @Test
-    void thatReturnInputStream() throws IOException {
-        when(storageSourceService.findSongBySource(any(Source.class))).thenReturn(inputStream);
-        storageSourceService.findSongBySource(source1);
-        verify(storageSourceService, times(1)).findSongBySource(any(Source.class));
+        verify(storageSourceService, times(0)).save(mockMultipartFile.getInputStream()
```

Comment:
Minor: you can use Mockito.never() verification mode instead of times(0), looks better in my opinion. But this is up to you what mode to use in such verifications.